ui/cap-react/src/components/drafts/form/themes/grommet/widgets/TextWidget.js

```diff
@@ -284,7 +282,17 @@ class TextWidget extends Component {
             this.props.options &&
             this.props.options.warningMessage
           }
+          mask={this.props.options.masked_array}
+          schemaMask={this.props.schema.pattern}
```

Comment:
(1) if `mask array` is passed, we should provide UX help (by masking the input) and leave the validation to the regex pattern (if exists). (2) if pattern is passed, it is used only for validation ( frontend, backend)